src/Compilers/CSharp/Test/Syntax/Diagnostics/DiagnosticTest.cs

```diff
@@ -102,6 +104,31 @@ public void TestCustomErrorInfo()
             Assert.Equal("OtherSymbol", (d3.Info as CustomErrorInfo).OtherSymbol);
         }
 
+        [Fact, WorkItem(66037, "https://github.com/dotnet/roslyn/issues/66037")]
+        public void DiagnosticInfo_WithSeverity()
+        {
+            var comp = CreateCompilation("");
+            var args = new object[] { comp.GlobalNamespace };
+            var symbol = (Symbol)comp.GlobalNamespace;
+            var type = TypeWithAnnotations.Create(comp.GetSpecialType(SpecialType.System_Object));
+
+            verifyWithSeverity(new CSDiagnosticInfo(ErrorCode.ERR_AbstractField));
+            verifyWithSeverity(new DiagnosticInfoWithSymbols(ErrorCode.ERR_DuplicateTypeParameter, args,
+                ImmutableArray.Create(symbol)));
+            verifyWithSeverity(new LazyArrayElementCantBeRefAnyDiagnosticInfo(type));
+            verifyWithSeverity(new LazyObsoleteDiagnosticInfo(symbol, symbol, BinderFlags.None));
+            verifyWithSeverity(new LazyUseSiteDiagnosticsInfoForNullableType(LanguageVersion.CSharp11, type));
+            verifyWithSeverity(new SyntaxDiagnosticInfo(1, 2, ErrorCode.ERR_DuplicateTypeParameter, args));
+            verifyWithSeverity(new XmlSyntaxDiagnosticInfo(XmlParseErrorCode.XML_EndTagExpected, args));
+
+            static void verifyWithSeverity(DiagnosticInfo diagnostic)
+            {
+                var other = diagnostic.GetInstanceWithSeverity(DiagnosticSeverity.Info);
```

Comment:
nit: you could also verify that the instance remains identical when the severity isn't different